states/battle.py

```diff
@@ -28,30 +30,126 @@ class Battle(State):
     def __init__(self):
         super().__init__()
         self.enemy = asyncio.run(pokeapi.get_random_pokemon_from_api())
-        is_escape = False
+        end_encounter = False
+
         print(MESSAGES["BATTLE"]["ENTRY"].format(self.enemy.name), end="\n" * 2)
-        print(f"Go {PLAYER.lead_pokemon.name}! \n")
-        while is_escape is not True:
+        self.send_pokemon(PLAYER.lead_pokemon)
+        self.print_health(PLAYER.lead_pokemon)
+
+        while end_encounter is not True:
             self.option = self.check_input(BattleOptions)
             # TODO: Implement battling
             if self.option == BattleOptions.FIGHT:
                 pass
             elif self.option == BattleOptions.TEAM:
                 pass
             elif self.option == BattleOptions.CATCH:
-                pass
+                if self.calc_catch_prob() is True:
```

Comment:
Consider splitting this out into a void function

Reply:
Created `catch_fail()` to handle that functionality